src/components/WelcomeFilters.tsx

```diff
@@ -17,6 +22,14 @@ export const WelcomeFilters: FC<{
   const texts = useTexts()
   const { push } = useRouter()
   const isMobile = useIsMobile()
+  const [urlState] = useUrlState()
+  const labels = useLabels()
+  const { latitude, longitude } = useUserGeolocation()
+  const tags = urlState.tags || []
+
+  const filteredRecords = recordsWithOnlyLabels.filter((r) =>
+    tags.every((f) => r.find((id) => id === f))
```

Comment:
Here the same comment as above concerning the variable names.

Reply:
Very good point! I renamed the variables!